### DISCUSSION TITLE
Applying acrylic effect to Win32 context menus

### DISCUSSION THREAD

**Discussion by krlvm** [valinet/ExplorerPatcher#292] [Ideas]
Hello,

It would be nice to see acrylic effect in the legacy Win32 context menus after disabling the brand new, slow immersive explorer context menu. It's somehow implemented in StartAllBack, but it is hardly possible to reverse-engineer it I think.

I'd like to contribute this feature and I've some success in developing it from scratch, but it currently has some problems, the main of which is a forced delay before displaying blur effect.

With [suggested implementation](https://github.com/krlvm/AcrylicMenus), acrylic effect can be shown on any Win32 context menu with using only `SetWindowsHookEx`:
![win11-2](https://user-images.githubusercontent.com/51774833/140634921-fee4a202-3ad8-40ce-a05f-257cd50c30af.png)

Maybe someone knows how to fix this?

**Comment by valinet** [comment]
https://github.com/krlvm/AcrylicMenus/issues/1#issuecomment-962880657
